src/server/detail/save_stages_controller.cc

```diff
@@ -378,7 +378,7 @@ GenericError SaveStagesController::FinalizeFileMovement() {
 
 // Build full path: get dir, try creating dirs, get filename with placeholder
 GenericError SaveStagesController::BuildFullPath() {
-  fs::path dir_path = GetFlag(FLAGS_dir);
+  fs::path dir_path = fs::path(GetFlag(FLAGS_dir)) / sub_path_;
```

Comment:
I have a small concern here. So how it works now, is that `FLAGS_dir` acts as the `base` directory to save on s3. So for example if `dir == s3://myfolder` . Then `SAVE df myname` will save it on s3 at `s3://myfolder/myname.dfs` 

Now with your changes if I do: `SAVE df s3://kostas/myname` it will save to `s3://myfolder/s3://kostas/myname.dfs` which I am think a) it's invalid path and b) is wrong because I want the path to be `s3://kostas/` and not under the base dir `s3://FLAGS_dir/kostas/`. 

@Abhra303 Can you confirm the behavior you want ? So let's say if:

`dir = s3://myfolder` and you call `SAVE df s3://kostas/mysnap` it will save to `s3://kostas/` ignoring the `base dir` correct ?

Reply:
We don't have specification how this should be done, so current implementation is done in a way that you specify only sub paths / basename for save command (this should work for all cloud providers and filesystem in same way). Of course, if we want to have feature that works in a way that you are expecting - i would modify it with next commits.

Also, fyi your example is valid path (on S3/Filesystem).

Reply:
I understand, basically I was trying to say what Roman wrote below :smile: 

p.s. I did not even my example was valid path -- my knowledge of s3 is fairly limited